src/web/components/Input/RadioInput.spec.tsx

```diff
@@ -0,0 +1,40 @@
+import { composeStories } from '@storybook/testing-react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import * as stories from './RadioInput.stories';
+
+const { WithLabel, WithValidation } = composeStories(stories);
+
+describe('RadioInput', () => {
+  it('renders correctly with label', () => {
+    render(<WithLabel />);
+    expect(screen.getByText('Select an option')).toBeInTheDocument();
+    expect(screen.getByText('Option 1')).toBeInTheDocument();
+    expect(screen.getByText('Option 2')).toBeInTheDocument();
+    expect(screen.getByText('Option 3')).toBeInTheDocument();
+  });
+
+  it('displays validation error message', async () => {
```

Comment:
I'm noticing a lot of identical tests - to me this is a sign something isn't quite right. I'll drop you a DM with some more ideas on this.